### PR TITLE
Capturing system-level metrics

### PR DESCRIPTION
**Pull request by vishnuchalla**
### Description
JIRA: https://issues.redhat.com/browse/OLS-769?filter=-1
Adding code snippets to capture system level metrics during and after the load test.

### Testing
Tested and verified in local also with elastic search. Please take a look at the attached metrics
[collected-metrics-1b696cd6-e3f2-4b0e-895d-59bb243f5579.zip](https://github.com/user-attachments/files/16135783/collected-metrics-1b696cd6-e3f2-4b0e-895d-59bb243f5579.zip)

### CODE REVIEW

**Comment by vishnuchalla** [comment]
cc: @xrajesh @tisnik @syedriko @JoaoFula

**Comment by JoaoFula** [comment]
lgtm but I'm not a go expert.

**Review by tisnik — COMMENTED** [review]
looks sane

**Review by tisnik — COMMENTED** [review]
what the `vegeta` and `attacker` means in terms of load generator?

**Comment by vishnuchalla** [comment]
> what the `vegeta` and `attacker` means in terms of load generator?

`vegeta` is a http benchmarking tool that has a go-client library to use. `attacker` is our own wrapper on top of vegeta to send load to the desired endpoints.

**Comment by vishnuchalla** [comment]
@xrajesh @tisnik Can we get this PR merged please? I would need these changes for load testing.

File: Makefile [hunk 1]
```diff
@@ -45,6 +45,8 @@ install:
 
 images:
 	@echo -e "\033[2mBuilding container $(CONTAINER_NAME_ARCH)\033[0m"
+	mkdir -p $(BIN_DIR)/$(ARCH)/assets/profiles
```
Comment (xrajesh) [inline Makefile:48]: @vishnuchalla Not related to this PR - but `ORG ?= vchalla` be updated to `openshift-lightspeed`

File: config/ols-load-generator.yaml [hunk 2]
```diff
@@ -40,18 +40,34 @@ spec:
         env:
           - name: OLS_TEST_HOST
             value: <ols-host-url>
-          - name: OLS_TEST_RUNID
-            value: <ols-test-runid>
+          - name: OLS_TEST_UUID
```
Comment (xrajesh) [inline config/ols-load-generator.yaml:43]: @vishnuchalla here too the image has reference to personal repo. ( not part of this PR )

File: README.md [hunk 3]
```diff
@@ -17,16 +17,28 @@ If running on openshift simply run `make` which will build and push the image to
...
 * `OLS_TEST_AUTH_TOKEN` - OLS auth token string.
 * `OLS_TEST_HIT_SIZE` - Indicates the total amount of requests to be sent as a part of the load testing.
 * `OLS_TEST_RPS` - Requests per second to be sent to the OLS endpoint in parallel.
+* `OLS_TEST_METRIC_STEP` - Step size for the cluster prometheus metrics to be captured.
+* `OLS_TEST_PROFILES` - List of metric profiles that contain queries to be executed on prometheus.
+* `OLS_TEST_ES_HOST`(Optional) - Elastic search host url. If not specified metrics will be indexed locally.
+* `OLS_TEST_ES_INDEX`(Optional) - Elastic search index name to store the data. If not specified metrics will be indexed locally.
 
-### **Example Usage**
+#### **Example Usage**
```
Comment (xrajesh) [inline README.md:41]: The below line should be ~/config/ols-load-generator.yaml ?

Comment (vishnuchalla) [reply README.md:41]: Updated as suggested.

File: config/ols-load-generator.yaml (hunk 2 above)
Comment (vishnuchalla) [reply config/ols-load-generator.yaml:43]: Yes, will take them in a separate PR. Once we have things setup in prow. Open PR: https://github.com/openshift/release/pull/53601

File: README.md [hunk 4]
```diff
@@ -17,18 +17,30 @@ If running on openshift simply run `make` which will build and push the image to
 
 ## **Usage**
 ### **Usage on openshift platform**
-Once we have OLS deployed and running on openshift cluster, in order to trigger the load test, deploy [assets/ols-load-generator.yaml](https://github.com/vishnuchalla/ols-load-generator/blob/perf-testing/assets/ols-load-generator.yaml) replaced with your corresponding envs values onto the cluster.
+Once we have OLS deployed and running on openshift cluster, in order to trigger the load test follow the below steps.
 
-### **Envs**
+
```
Comment (xrajesh) [inline README.md:22]: @vishnuchalla before we create the secret the - namespace creation is needed. It will be nice to add here .

**Comment by vishnuchalla** [comment]
@xrajesh @tisnik Can we get this PR merged please, if the changes look okay? We can work on further improvements in the follow-up PRs as we finalize on the second iteration of the test plan.
